dungeon/src/contrib/crafting/Crafting.java

```diff
@@ -80,57 +79,26 @@ public static void clearRecipes() {
    * <p>If the program is compiled to a jar file, recipes will be loaded from within the jar file.
    */
   public static void loadRecipes() {
-    if (Objects.requireNonNull(Crafting.class.getResource("/recipes"))
-        .toString()
-        .startsWith("jar:")) {
-      loadFromJar();
-    } else {
-      loadFromFile();
-    }
-  }
-
-  /** Load recipes if the program was started from a jar file. */
-  private static void loadFromJar() {
+    final String dirName = "recipes/";
     try {
-      String path =
-          new File(Objects.requireNonNull(Game.class.getResource("")).getPath())
-              .getParent()
-              // for windows
-              .replaceAll("(!|file:\\\\)", "")
-              // for unix/macos
-              .replaceAll("(!|file:)", "");
-      JarFile jar = new JarFile(path);
-      Enumeration<JarEntry> entries = jar.entries();
-      while (entries.hasMoreElements()) {
-        JarEntry entry = entries.nextElement();
-        if (entry.getName().startsWith("recipes") && entry.getName().endsWith(".recipe")) {
-          LOGGER.info("Load recipe: " + entry.getName());
-          Recipe r =
-              parseRecipe(Game.class.getResourceAsStream("/" + entry.getName()), entry.getName());
-          if (r != null) Crafting.RECIPES.add(r);
-        }
-      }
-    } catch (IOException e) {
-      throw new RuntimeException(e);
-    }
-  }
-
-  /** Load recipes if the program was started from a folder. */
-  private static void loadFromFile() {
-    File folder =
-        new File(Objects.requireNonNull(Crafting.class.getResource("/recipes")).getPath());
-    File[] files = folder.listFiles();
-    if (files == null) {
-      return;
-    }
-    for (File file : files) {
-      if (file.getName().endsWith(".recipe")) {
-        LOGGER.info("Load recipe: " + file.getName());
-        Recipe r =
-            parseRecipe(
-                Crafting.class.getResourceAsStream("/recipes/" + file.getName()), file.getName());
-        if (r != null) Crafting.RECIPES.add(r);
-      }
+      core.components.FileSystemUtil.visitResources(
```

Comment:
Warum nicht `FileSystemUtil`, warum hier fully qualified?!

Reply:
Danke fürs Review. Ich schaue mir alle Punkte noch einmal in Ruhe an.

Speziell hier geht es auch mit dem normalen `import`. Bin mir unsicher, was ich dabei gedacht habe - aber die Paketstruktur an dieser Stelle explizit hervorzuheben, und damit vom gewohnten "Schema" abzuweichen, macht wenig Sinn, ja.